mesonbuild/modules/i18n.py

```diff
@@ -256,10 +256,18 @@ def gettext(self, state: 'ModuleState', args: T.Tuple[str], kwargs: 'Gettext') -
 
         extra_arg = '--extra-args=' + '@@'.join(extra_args) if extra_args else None
 
+        if state.subproject:
+            source_root = path.join(state.environment.source_dir, self.interpreter.subproject_dir, state.subproject)
```

Comment:
The name of the subproject is not necessarily the name of its directory.

IIRC you can use interpreter.root_subdir for this.

Reply:
That's... entirely possible lol.

If it's wrong for one location it should be wrong for both of them. But documentation builds aren't all that more common than maintainer builds when dealing with subprojects.

Reply:
Yep, you can confirm this if you try to build a hotdoc subproject and look at `subprojects/foobar-1.0/*-doc.json` and observe the `"gi_c_source_roots"` key.

It does indeed need fixing, want to fix it as a followup commit in this PR?

Reply:
Wait, lol. Even worse. If it is NOT a subproject, this caused `"gi_c_source_roots"` to be e.g.

```
[
    "path/to/meson/docs/subprojects/",
    "path/to/meson/docs/builddir/subprojects/"
]
```

Which seems excessively broken...

Reply:
@thiblahute This seems probably broken from commit c6012e3cc728982797e794fd28b89f1ff3bb25cc and b4347ca4a98b148ba7f7908c3d96f393294df122?